src/components/publishing/display/canvas/slideshow.tsx

```diff
@@ -0,0 +1,217 @@
+    )
+  }
+
+  renderSlides = () => {
+    const { unit, containerWidth } = this.props
+    return unit.assets.map((image, i) => {
+      return (
+        <Slide key={i} className={i === 0 && "title-card"} containerWidth={containerWidth}>
+          {i === 0 && containerWidth >= 900 && this.renderTitleCard()}
+          <div>
+            <Image
+              src={crop(image.url, { width: 780, height: 460 })}
+              alt={image.caption || ""}
+              containerWidth={containerWidth}
+            />
+            {image.caption && <Caption>{image.caption}</Caption>}
+          </div>
+        </Slide>
+      )
+    })
+  }
+
+  onSlideChange = currentSlide => {
+    this.setState({ isOnTitle: currentSlide === 0 })
+  }
+
+  render() {
+    const { containerWidth } = this.props
+    const sliderSettings = {
+      dots: false,
+      lazyLoad: false,
+      infinite: false,
+      variableWidth: true,
+      centerMode: true,
+      nextArrow: <RightArrow containerWidth={containerWidth} onRightArrow={() => this.slider.slickGoTo(0)} />,
+      prevArrow: <LeftArrow containerWidth={containerWidth} isOnTitle={this.state.isOnTitle} />,
+      initialSlide: 0,
+      afterChange: currentSlide => {
+        this.onSlideChange(currentSlide)
+      },
+      responsive: [
+        {
+          breakpoint: 900,
+          settings: {
+            infinite: true,
+          },
+        },
+      ],
+    }
+
+    return (
+      <SliderContainer containerWidth={containerWidth}>
+        <Slider {...sliderSettings} ref={slider => (this.slider = slider)}>
+          {this.renderSlides()}
+        </Slider>
+        {containerWidth < 900 && this.renderTitleCard()}
+      </SliderContainer>
+    )
+  }
+}
+
+const LeftArrow = props => {
+  return (
+    <NavArrow
+      containerWidth={props.containerWidth}
+      direction="left"
+      onClick={props.onClick}
+      isVisible={!props.isOnTitle}
+    >
+      <Icon name="chevron-left" color="black" fontSize="24px" />
+    </NavArrow>
+  )
+}
+
+const RightArrow = props => {
+  const onClick = props.currentSlide + 1 === props.slideCount ? props.onRightArrow : props.onClick
+  return (
+    <NavArrow containerWidth={props.containerWidth} direction="right" onClick={onClick} isVisible>
+      <Icon name="chevron-right" color="black" fontSize="24px" />
+    </NavArrow>
+  )
+}
+
+interface NavArrowProps extends React.HTMLProps<HTMLDivElement> {
+  direction: string
+  isVisible?: boolean
+  containerWidth: number
+}
+
+interface ResponsiveProps extends React.HTMLProps<HTMLDivElement> {
+  containerWidth: number
+}
+
+const arrowDiv: StyledFunction<NavArrowProps> = styled.div
+const responsiveDiv: StyledFunction<ResponsiveProps> = styled.div
+const responsiveImage: StyledFunction<ResponsiveProps> = styled.img
+
+const NavArrow = arrowDiv`
+  display: flex;
+  align-items: center;
+  position: absolute;
+  height: 100%;
+  max-height: ${props => props.containerWidth * 0.65 * 0.59 + "px;"}
```

Comment:
These values seem arbitrary to me -- what are they representing? Maybe we can extract some mystery by making them constants.

Reply:
A constant might be clearer here -- .65 is the max width of an image, .59 is the aspect ratio.

Reply:
Ok yeah, I see it being used throughout the file so a constant makes sense here.